lib/Stow/Util.pm.in

```diff
@@ -193,6 +193,20 @@ sub restore_cwd {
     chdir($prev) or error("Your current directory $prev seems to have vanished");
 }
 
+sub adjust_dotfile {
+    my $target = $_[0];
+
+    my @result = ();
+    for my $part (split m{/+}, $target) {
+        if ($part ne "dot-") {
```

Comment:
I like the attention to detail here :-)  I think you need to also defend against `dot-.` though, as that would currently be turned into `..`.